### PR TITLE
internal: Newtype `ErasedFileAstId`

### PR DESCRIPTION
**Pull request by Veykril**
It wrapping `la_arena::Idx` makes it quite annoying to use

### CODE REVIEW

File: crates/project-model/src/workspace.rs [hunk 1]
```diff
@@ -1349,12 +1349,13 @@ fn add_target_crate_root(
     );
     if let TargetKind::Lib { is_proc_macro: true } = kind {
         let proc_macro = match build_data.as_ref().map(|it| it.proc_macro_dylib_path.as_ref()) {
-            Some(it) => it.cloned().map(|path| Ok((cargo_name.to_owned(), path))),
-            None => Some(Err("proc-macro crate is missing its build data".to_owned())),
+            Some(it) => match it {
+                Some(path) => Ok((cargo_name.to_owned(), path.clone())),
+                None => Err("proc-macro crate build data is missing dylib path".to_owned()),
```
Comment (bugadani) [inline crates/project-model/src/workspace.rs:1354]: What may cause me hitting this error on most proc macros?

Comment (Veykril) [reply crates/project-model/src/workspace.rs:1354]: Unclear, somethings going wrong when cargo is reporting the build artifact outputs then (or your build script command is configured to something else that misbehaves)

**Pull request merged**